src/Command/HelpCommand.php

```diff
@@ -95,23 +97,34 @@ protected function asText($io, $commands)
             $invert[$class][] = $name;
         }
 
-        foreach ($commands as $name => $class) {
-            if (is_object($class)) {
-                $class = get_class($class);
+        $prefixed = [];
+        foreach ($invert as $class => $names) {
+            $prefixedName = $this->findPrefixedName($names);
+            if (!$prefixedName) {
+                $prefix = preg_match('#^Cake\\\\(Command|Shell)\\\\#', $class) ? '[core]' : '[app]';
```

Comment:
Do you need `Command|Shell` starting with `Cake` should be enough to flag as CakePHP/App. I also wonder if `[core]` should just be `CakePHP`.

Reply:
People have their own plugins prefixed like this ;) Not nice, but the case:

CakeExtra plugin:

    Cake\Extra\Command\...

Reply:
I didn't think of that :+1: